app.rb

```diff
@@ -0,0 +1,103 @@
+require_relative './person_class'
+require_relative './book'
+require_relative './rental'
+require_relative './teacher_class'
+require_relative './student_class'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'Books:'
+    @books.each { |book| puts book }
+  end
+
+  def list_all_people
+    puts 'People:'
+    @people.each { |person| puts person }
+  end
+
+  def create_person_prompt
+    puts 'Enter person name:'
+    name = gets.chomp
+
+    puts 'Enter person age:'
+    age = gets.chomp.to_i
+
+    puts 'Enter person type (teacher or student):'
+    type = gets.chomp
+
+    create_person(name, age, type)
+  end
+
+  def create_person(name, age, type)
+    person = type == 'teacher' ? Teacher.new(name, age) : Student.new(name, age)
+    @people << person
```

Comment:
Let's ensure to resolve the below:
- [ ] The app crashes in an attempt to create a _person_, either of type _teacher_ or _student_. 
- [ ] Please let's be reminded that the _person_ of type _teacher_ needs not to have the Parent's permission property. 

**Creating a person of type student error:**
![image](https://github.com/Tracy-miranja/school_library/assets/12745474/6ce7dbe9-08ba-47e5-a440-88d7e510799f)


**Creating a person of type student error:**
![image](https://github.com/Tracy-miranja/school_library/assets/12745474/5acf4081-4080-481a-a283-ea1f36683e7f)